### PR TITLE
Fixes #22616 - Allow searching hosts by status as integer

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Comment by tbrisker** [comment]
This was introduced by 3adfb7d1b0338bfa26232b99037a13f9bf4862e9 which breaks built-in searches, once merged it should be cherry-picked to 1.17 as well

**Comment by lzap** [comment]
What is the query to test?

**Comment by tbrisker** [comment]
@lzap most of the queries on the dashboard config status widget and the default bookmarks on the hosts index

**Comment by lzap** [comment]
Yeah I mean what should I type into the search bar, I am trying with `status = ` but it does not work. Is this overall status (like build status) or report status?

**Comment by tbrisker** [comment]
`status.{enabled,applied,...} = 0`

File: test/models/host_status/configuration_status_test.rb [hunk 1]
```diff
@@ -115,4 +115,10 @@ def setup
     assert_equal '6 & 63', HostStatus::ConfigurationStatus.bit_mask('restarted')
     assert_equal '12 & 63', HostStatus::ConfigurationStatus.bit_mask('failed')
   end
+
+  test 'host search by status works' do
+    @status.save
+    assert_equal Host.search_for('status.applied = 0'), [@host]
+    assert_equal Host.search_for('status.applied = false'), [@host]
+  end
```
Comment (lzap) [inline test/models/host_status/configuration_status_test.rb]: Expected value should be the first parameter. Can you also add a positive test when there is a number greater than zero?

**Review by lzap — CHANGES_REQUESTED** [review]
Minor comment but works fine.

File: test/models/host_status/configuration_status_test.rb (hunk 1 above)
Comment (tbrisker) [reply test/models/host_status/configuration_status_test.rb]: Updated, thanks!

**Review by lzap — APPROVED** [review]
Thanks. Waiting for Mr. Jenkins.

**Pull request merged**